lxd/instance/drivers/driver_qemu.go

```diff
@@ -3215,29 +3210,43 @@ func (d *qemu) addNetDevConfig(cpuCount int, busName string, qemuDev map[string]
 	return nil, fmt.Errorf("Unrecognised device type")
 }
 
-func (d *qemu) writeNICDevConfig(mtuStr string, devName string, devHwaddr string) error {
-	mtuInt, err := strconv.Atoi(mtuStr)
+// writeNICDevConfig writes the NIC config for the specified device into the NICConfigDir.
+// This will be used by the lxd-agent to rename the NIC interfaces inside the VM guest.
+func (d *qemu) writeNICDevConfig(mtuStr string, devName string, nicName string, devHwaddr string) error {
+	// Parse MAC address to ensure it is in a canonical form (avoiding casing/presentation differences).
+	hw, err := net.ParseMAC(devHwaddr)
 	if err != nil {
-		logger.Warn("Invalid MTU returned by device run configuration", log.Ctx{
-			"deviceName": devName,
-		})
-		mtuInt = 0
+		return fmt.Errorf("Failed parsing MAC: %w", err)
 	}
-	mtu := uint32(mtuInt)
 
-	// Parse MAC address to ensure it is in a consistent form (avoiding casing errors).
-	hw, err := net.ParseMAC(devHwaddr)
+	nicConfig := deviceConfig.NICConfig{
+		DeviceName: devName,
+		NICName:    nicName,
+		MACAddress: hw.String(),
+	}
+
+	if mtuStr != "" {
+		mtuInt, err := strconv.ParseUint(mtuStr, 10, 32)
+		if err != nil {
+			return fmt.Errorf("Failed parsing MTU: %w", err)
+		}
+
+		nicConfig.MTU = uint32(mtuInt)
+	}
+
+	nicConfigBytes, err := json.Marshal(nicConfig)
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed encoding NIC config: %w", err)
 	}
 
-	nicConfigBytes, err := json.Marshal(deviceConfig.NICConfig{MACAddress: hw.String(), MTU: mtu})
+	nicFile := filepath.Join(d.Path(), "config", deviceConfig.NICConfigDir, fmt.Sprintf("%s.json", filesystem.PathNameEncode(nicConfig.DeviceName)))
```

Comment:
Same as https://github.com/lxc/lxd/pull/9967/files#r813838445 I'm not sure I understand the need to encode the device name here as it shouldn't contain characters unfit for a filename.